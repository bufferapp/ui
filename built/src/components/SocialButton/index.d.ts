export { default } from './SocialButton';
//# sourceMappingURL=index.d.ts.map