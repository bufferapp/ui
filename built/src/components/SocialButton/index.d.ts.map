{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/SocialButton/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,gBAAgB,CAAA"}