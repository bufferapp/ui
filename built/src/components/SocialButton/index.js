export { default } from './SocialButton';
//# sourceMappingURL=index.js.map