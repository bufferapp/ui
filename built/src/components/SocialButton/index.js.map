{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/SocialButton/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,gBAAgB,CAAA","sourcesContent":["export { default } from './SocialButton'\n"]}