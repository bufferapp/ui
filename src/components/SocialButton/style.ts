import styled from 'styled-components'
import {
  facebookDark,
  instagramDark,
  linkedinDark,
  pinterestDark,
  twitterDark,
  white,
} from '../style/colors'
import { ButtonBase, large } from '../Button/style'

// @ts-expect-error TS(7006) FIXME: Parameter 'channel' implicitly has an 'any' type.
function calculateBgColor(channel) {
  switch (channel) {
    case 'instagram':
      return instagramDark
    case 'twitter':
      return twitterDark
    case 'facebook':
      return facebookDark
    case 'pinterest':
      return pinterestDark
    case 'linkedin':
      return linkedinDark
    default:
      return 'gray'
  }
}

export const SocialButtonStyled = styled.button`
  ${ButtonBase};
  ${large};
  background-color: ${(props) => calculateBgColor(props.channel)};
  color: ${white};
  width: 100%;
  justify-content: space-between;
  cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
  box-shadow: 0px 2px 10px rgba(0, 0, 0, 0);
  transition: box-shadow 0.1s ease;

  p {
    color: ${white};
    margin-left: 8px;
    line-height: normal;
  }

  &:hover {
    box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.25);
    transition: box-shadow 0.1s ease;
  }
`

export const ChannelIconContainerStyled = styled.span`
  display: flex;
  align-items: center;
`
