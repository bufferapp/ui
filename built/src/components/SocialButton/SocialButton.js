import React from 'react';
import PropTypes from 'prop-types';
import { Twitter, Instagram, Facebook, Pinterest, LinkedIn } from '../Icon';
import Text from '../Text/Text';
import { ChannelIconContainerStyled, SocialButtonStyled } from './style';
const renderContent = (channel) => {
    const content = {};
    switch (channel) {
        case 'instagram':
            content.Icon = React.createElement(Instagram, null);
            content.name = 'Instagram';
            content.cta = 'Connect an Instagram business account';
            break;
        case 'twitter':
            content.Icon = React.createElement(Twitter, null);
            content.name = 'Twitter';
            content.cta = 'Connect a Twitter profile';
            break;
        case 'facebook':
            content.Icon = React.createElement(Facebook, null);
            content.name = 'Facebook';
            content.cta = 'Connect a Facebook page';
            break;
        case 'pinterest':
            content.Icon = React.createElement(Pinterest, null);
            content.name = 'Pinterest';
            content.cta = 'Connect a Pinterest profile';
            break;
        case 'linkedin':
            content.Icon = React.createElement(LinkedIn, null);
            content.name = 'LinkedIn';
            content.cta = 'Connect a LinkedIn profile';
            break;
        default:
            break;
    }
    return content;
};
const SocialButton = ({ disabled, channel, onClick }) => {
    const content = renderContent(channel);
    return (React.createElement(SocialButtonStyled, { onClick: !disabled ? onClick : undefined, channel: channel, disabled: disabled },
        React.createElement(ChannelIconContainerStyled, null,
            content.Icon,
            React.createElement(Text, { type: "p" }, content.name)),
        React.createElement(Text, { type: "p" }, content.cta)));
};
SocialButton.propTypes = {
    disabled: PropTypes.bool,
    channel: PropTypes.oneOf([
        'twitter',
        'instagram',
        'facebook',
        'pinterest',
        'linkedin',
    ]).isRequired,
    onClick: PropTypes.func.isRequired,
};
SocialButton.defaultProps = {
    disabled: false,
};
export default SocialButton;
//# sourceMappingURL=SocialButton.js.map