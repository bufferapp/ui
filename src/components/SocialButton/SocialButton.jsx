import React from 'react'
import PropTypes from 'prop-types'
import { Twitter, Instagram, Facebook, Pinterest, LinkedIn } from '../Icon'
import Text from '../Text/Text'

import { ChannelIconContainerStyled, SocialButtonStyled } from './style'

const renderContent = (channel) => {
  const content = {}
  switch (channel) {
    case 'instagram':
      content.Icon = <Instagram />
      content.name = 'Instagram'
      content.cta = 'Connect an Instagram business account'
      break
    case 'twitter':
      content.Icon = <Twitter />
      content.name = 'Twitter'
      content.cta = 'Connect a Twitter profile'
      break
    case 'facebook':
      content.Icon = <Facebook />
      content.name = 'Facebook'
      content.cta = 'Connect a Facebook page'
      break
    case 'pinterest':
      content.Icon = <Pinterest />
      content.name = 'Pinterest'
      content.cta = 'Connect a Pinterest profile'
      break
    case 'linkedin':
      content.Icon = <LinkedIn />
      content.name = 'LinkedIn'
      content.cta = 'Connect a LinkedIn profile'
      break
    default:
      break
  }
  return content
}

const SocialButton = ({ disabled, channel, onClick }) => {
  const content = renderContent(channel)
  return (
    <SocialButtonStyled
      onClick={!disabled ? onClick : undefined}
      channel={channel}
      disabled={disabled}
    >
      <ChannelIconContainerStyled>
        {content.Icon}
        <Text type="p">{content.name}</Text>
      </ChannelIconContainerStyled>
      <Text type="p">{content.cta}</Text>
    </SocialButtonStyled>
  )
}

SocialButton.propTypes = {
  /** Is the button disabled */
  disabled: PropTypes.bool,
  /** Channel */
  channel: PropTypes.oneOf([
    'twitter',
    'instagram',
    'facebook',
    'pinterest',
    'linkedin',
  ]).isRequired,
  /** OnClick handler */
  onClick: PropTypes.func.isRequired,
}

SocialButton.defaultProps = {
  disabled: false,
}

export default SocialButton
