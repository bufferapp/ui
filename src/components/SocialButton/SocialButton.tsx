import React from 'react'
import PropTypes from 'prop-types'
import { Twitter, Instagram, Facebook, Pinterest, LinkedIn } from '../Icon'
import Text from '../Text/Text'

import { ChannelIconContainerStyled, SocialButtonStyled } from './style'

// @ts-expect-error TS(7006) FIXME: Parameter 'channel' implicitly has an 'any' type.
const renderContent = (channel) => {
  const content = {}
  switch (channel) {
    case 'instagram':
      // @ts-expect-error TS(2339) FIXME: Property 'Icon' does not exist on type '{}'.
      content.Icon = <Instagram />
      // @ts-expect-error TS(2339) FIXME: Property 'name' does not exist on type '{}'.
      content.name = 'Instagram'
      // @ts-expect-error TS(2339) FIXME: Property 'cta' does not exist on type '{}'.
      content.cta = 'Connect an Instagram business account'
      break
    case 'twitter':
      // @ts-expect-error TS(2339) FIXME: Property 'Icon' does not exist on type '{}'.
      content.Icon = <Twitter />
      // @ts-expect-error TS(2339) FIXME: Property 'name' does not exist on type '{}'.
      content.name = 'Twitter'
      // @ts-expect-error TS(2339) FIXME: Property 'cta' does not exist on type '{}'.
      content.cta = 'Connect a Twitter profile'
      break
    case 'facebook':
      // @ts-expect-error TS(2339) FIXME: Property 'Icon' does not exist on type '{}'.
      content.Icon = <Facebook />
      // @ts-expect-error TS(2339) FIXME: Property 'name' does not exist on type '{}'.
      content.name = 'Facebook'
      // @ts-expect-error TS(2339) FIXME: Property 'cta' does not exist on type '{}'.
      content.cta = 'Connect a Facebook page'
      break
    case 'pinterest':
      // @ts-expect-error TS(2339) FIXME: Property 'Icon' does not exist on type '{}'.
      content.Icon = <Pinterest />
      // @ts-expect-error TS(2339) FIXME: Property 'name' does not exist on type '{}'.
      content.name = 'Pinterest'
      // @ts-expect-error TS(2339) FIXME: Property 'cta' does not exist on type '{}'.
      content.cta = 'Connect a Pinterest profile'
      break
    case 'linkedin':
      // @ts-expect-error TS(2339) FIXME: Property 'Icon' does not exist on type '{}'.
      content.Icon = <LinkedIn />
      // @ts-expect-error TS(2339) FIXME: Property 'name' does not exist on type '{}'.
      content.name = 'LinkedIn'
      // @ts-expect-error TS(2339) FIXME: Property 'cta' does not exist on type '{}'.
      content.cta = 'Connect a LinkedIn profile'
      break
    default:
      break
  }
  return content
}

// @ts-expect-error TS(7031) FIXME: Binding element 'disabled' implicitly has an 'any'... Remove this comment to see the full error message
const SocialButton = ({ disabled, channel, onClick }) => {
  const content = renderContent(channel)
  return (
    <SocialButtonStyled
      onClick={!disabled ? onClick : undefined}
      channel={channel}
      disabled={disabled}
    >
      <ChannelIconContainerStyled>
        {/* @ts-expect-error TS(2339) FIXME: Property 'Icon' does not exist on type '{}'. */}
        {content.Icon}
        {/* @ts-expect-error TS(2339) FIXME: Property 'name' does not exist on type '{}'. */}
        <Text type="p">{content.name}</Text>
      </ChannelIconContainerStyled>
      {/* @ts-expect-error TS(2339) FIXME: Property 'cta' does not exist on type '{}'. */}
      <Text type="p">{content.cta}</Text>
    </SocialButtonStyled>
  )
}

SocialButton.propTypes = {
  /** Is the button disabled */
  disabled: PropTypes.bool,
  /** Channel */
  channel: PropTypes.oneOf([
    'twitter',
    'instagram',
    'facebook',
    'pinterest',
    'linkedin',
  ]).isRequired,
  /** OnClick handler */
  onClick: PropTypes.func.isRequired,
}

SocialButton.defaultProps = {
  disabled: false,
}

export default SocialButton
