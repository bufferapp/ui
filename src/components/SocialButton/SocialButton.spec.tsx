/* eslint-disable react/jsx-filename-extension */
import React from 'react'
// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'jest... Remove this comment to see the full error message
import snap from 'jest-auto-snapshots'
// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'reac... Remove this comment to see the full error message
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import SocialButton from './SocialButton'

snap(SocialButton, './SocialButton.tsx')

test('Disabled button is disabled', () => {
  const tree = renderer
    .create(<SocialButton channel="twitter" onClick={() => {}} disabled />)
    .toJSON()
  expect(tree).toHaveStyleRule('cursor', 'not-allowed')
})
