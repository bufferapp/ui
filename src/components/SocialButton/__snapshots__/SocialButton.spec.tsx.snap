// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > SocialButton Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<styled.button
  channel="twitter"
  disabled={false}
  onClick={[MockFunction]}
>
  <styled.span>
    <TwitterIcon />
    <Text
      color="grayDarker"
      hasError={false}
      type="p"
    >
      Twitter
    </Text>
  </styled.span>
  <Text
    color="grayDarker"
    hasError={false}
    type="p"
  >
    Connect a Twitter profile
  </Text>
</styled.button>
`;

exports[`jest-auto-snapshots > SocialButton Matches snapshot when passed all props 1`] = `
<styled.button
  channel="twitter"
  disabled={true}
>
  <styled.span>
    <TwitterIcon />
    <Text
      color="grayDarker"
      hasError={false}
      type="p"
    >
      Twitter
    </Text>
  </styled.span>
  <Text
    color="grayDarker"
    hasError={false}
    type="p"
  >
    Connect a Twitter profile
  </Text>
</styled.button>
`;

exports[`jest-auto-snapshots > SocialButton Matches snapshot when passed only required props 1`] = `
<styled.button
  channel="twitter"
  disabled={false}
  onClick={[MockFunction]}
>
  <styled.span>
    <TwitterIcon />
    <Text
      color="grayDarker"
      hasError={false}
      type="p"
    >
      Twitter
    </Text>
  </styled.span>
  <Text
    color="grayDarker"
    hasError={false}
    type="p"
  >
    Connect a Twitter profile
  </Text>
</styled.button>
`;
